### PR TITLE
fix(settings): bitcoin address type menu

### PR DESCRIPTION
**Pull request by MiguelMedeiros**
## Description

This PR changes the UX when you select the default `bitcoin address type` on the `settings menu`.

**There is a delay when you change the address but it will be fixed by another PR which optimizes the performance to create a new bitcoin address.**

## Preview

https://user-images.githubusercontent.com/5798170/186425399-892444e9-bf6a-46de-af05-3e17928ab1c2.mp4

### CODE REVIEW

File: src/screens/Settings/AddressTypePreference/index.tsx [hunk 1]
```diff
@@ -81,7 +93,7 @@ const AddressTypeSettings = (): ReactElement => {
 
 	return (
 		<SettingsView
-			title={'Address types preference'}
+			title={'Bitcoin Address Types'}
```
Comment (pwltr) [inline src/screens/Settings/AddressTypePreference/index.tsx]: nit: should be "Bitcoin Address Type"

File: src/store/types/settings.ts [hunk 2]
```diff
@@ -38,6 +43,7 @@ export interface ISettings {
 	unitPreference: 'asset' | 'fiat';
 	showSuggestions: boolean;
 	transactionSpeed: TTransactionSpeed;
+	addressType: TAddressType;
 	[key: string]: any;
```
Comment (pwltr) [inline src/store/types/settings.ts]: Could remove this `[key: string]: any;` while we're here and fix the ensuing type errors

**Review by pwltr — APPROVED** [review]

File: src/screens/Settings/AddressTypePreference/index.tsx (hunk 1 above)
Comment (MiguelMedeiros) [reply src/screens/Settings/AddressTypePreference/index.tsx]: Done! [efd3043](https://github.com/synonymdev/bitkit/pull/280/commits/efd304323541383f551dc5c1f4160bda52c2d812)

File: src/store/types/settings.ts (hunk 2 above)
Comment (MiguelMedeiros) [reply src/store/types/settings.ts]: Done! [5a70860](https://github.com/synonymdev/bitkit/pull/280/commits/5a7086028175b789ae903ae9a5009a70f75d8efa)

File: src/screens/Settings/AddressTypePreference/index.tsx [hunk 3]
```diff
@@ -8,14 +9,19 @@ import { getSelectedAddressType } from '../../../utils/wallet';
 import { capitalize } from '../../../utils/helpers';
 import { updateSelectedAddressType } from '../../../store/actions/wallet';
 import { TAddressType } from '../../../store/types/wallet';
+import { updateSettings } from '../../../store/actions/settings';
 
 const AddressTypeSettings = (): ReactElement => {
+	const navigation = useNavigation();
```
Comment (limpbrains) [inline src/screens/Settings/AddressTypePreference/index.tsx]: navigation object is already in props, no need to use the hook
`const AddressTypeSettings = ({ navigation }): ReactElement => {`

File: src/store/types/settings.ts (hunk 2 above)
Comment (pwltr) [reply src/store/types/settings.ts]: Do you want to fix the 4 type errors that are now there as well? I know it's unrelated but we can clean up as we go. Up to you, I can also do it in another PR.

File: src/screens/Settings/AddressTypePreference/index.tsx (hunk 3 above)
Comment (MiguelMedeiros) [reply src/screens/Settings/AddressTypePreference/index.tsx]: Fixed here: [16901ae](https://github.com/synonymdev/bitkit/pull/280/commits/16901ae2fdad18a23f008d073f8488b4c1acb1d6)

Also changed the other files that used the same object.

**Review by limpbrains — APPROVED** [review]
Still a few typescript errors

**Pull request merged**